src/mm-input/mm-input.js

```diff
@@ -4,213 +4,237 @@
+				this.error = false;
+			} else {
+				this.error = true;
 			}
-			if (this.icon === "search" && this.search && this.clear || this.icon && this.clear) {
-				this.$.icon.style.display = "none";
+		},
+
+		validator: function() {
+			switch(this.validation) {
+				case "empty":
+					return this.value.length > 0;
+				case "int":
+					return this.inter.test(this.value);
+				case "decimal":
+					return this.floater.test(this.value);
+				case "email":
+					return this.email.test(this.value);
+				case "alpha":
+					return this.alpha.test(this.value);
+				default:
+					return true;
 			}
-			if (this.autocomplete) {
-				this.searchItems = this.data.filter(dataFilter.bind(this));
-				if (this.searchItems.length > 0) {
-					this.open();
+		},
+
+		updateUI: function(value) {
+			// handle the clear and search state styling:
+			if (value) {
+				if (this.clear) {
+					this.$.clearBtn.style.display = "block";
+				}
+				if (this.icon === "search" && this.search && this.clear || this.icon && this.clear) {
+					this.$.icon.style.display = "none";
 				}
-				// if (this.searchItems.length > this.maxItems) {
-				// this.searchItems = this.searchItems.splice(0, this.maxItems);
-				// }
-			}
-		} else {
-			if (this.clear) {
-				this.$.clearBtn.style.display = "none";
-			}
-			if (this.icon === "search" && this.search && this.clear || this.icon && this.clear) {
-				this.$.icon.style.display = "block";
-			}
-			if (this.autocomplete) {
-				this.close();
-			}
-		}
-	},
-	
-	bindModel: function(model, property) {
-		this.model = model;
-		this.property = property;
-	},
-
-	updateModel: function() {
-		if (this.model && this.property) {
-			//check for BB models
-			if (this.model.set) {
-				var o = {};
-				o[this.property] = this.value;
-				this.model.set(o);
 			} else {
-				this.model[this.property] = this.value;
+				if (this.clear) {
+					this.$.clearBtn.style.display = "none";
+				}
+				if (this.icon === "search" && this.search && this.clear || this.icon && this.clear) {
+					this.$.icon.style.display = "block";
+				}
 			}
+		},
+		
+		bindModel: function(model, property) {
+			this.model = model;
+			this.property = property;
+		},
+
+		updateModel: function() {
+			if (this.model && this.property) {
+				//check for BB models
+				if (this.model.set) {
+					var o = {};
+					o[this.property] = this.value;
+					this.model.set(o);
+				} else {
+					this.model[this.property] = this.value;
+				}
+			}
+		},
+
+		clearInput: function(e) {
+			this.value = "";
+			this.updateUI(false);
+		},
+
+		updateStyle: function(icon, width, fitparent, clear) {
+			var p = icon || clear ? this.PADDING_RIGHT_ICON + "px" : this.PADDING_RIGHT_DEFAULT + "px",
+				f = fitparent ? "100%" : false,
+				w = width ? width + "px" : "auto";
+
+			return this.styleBlock({
+				paddingRight: p,
+				width: f ? f : w
+			});
+		},
+
+		updateClass: function(state, error) {
+			var o = {};
+			o["text-input"] = true;
+			o["invalid"] = error;
```

Comment:
we only need [] notation for things that cant be represented as a js accessor ([] is slower)